core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandler.java

```diff
@@ -65,11 +69,9 @@ private Optional<Future<Void>> invokeBeforePreparationSnapshotHook(Snapshot snap
       return Optional.of(
           beforePreparationSnapshotHook.handle(tableMetadataManager, snapshot.getReadWriteSets()));
     } catch (Exception e) {
+      onFailureBeforeCommit(snapshot);
```

Comment:
For safety, I think we can catch the exception as follow:
```suggestion
      try {
        onFailureBeforeCommit(snapshot);
      } catch (Exception ex) {
        ...
      }
```

What do you think?

Reply:
Yeah, you have a point. On the other hand, I think all the codes calling the callback need the same just-in-case error handling in terms of consistency. So, I wrote this comment https://github.com/scalar-labs/scalardb/pull/2390/files#diff-80b71c10ff00c8a739d8aec837a1730b5470df99295bebb354033c9aefc33bbaR55-R56 so that we don't need such a just-in-case error handling. But if you think the comment isn't enough, I'm okay to add the error handling to all the codes calling the callback. Either is fine with me 👍

Reply:
@brfrn169 On second thought, I'll follow your suggestion since trusting a callback injected from outside is a bit too optimistic and the comment doesn't guarantee a callback won't throw an exception. Let me update the PR.